### PR TITLE
fix: payment term when creating PO from SO

### LINKED ISSUE #41369: When creating a Purchase Order from a Sales Order, supplier's default payment terms template isn't being used
### Information about bug

When we create a PO from a Sales Order (`Create` -> `Purchase Order` inside Sales Order doc), the Sales Order's terms template and payment_schedule gets copied to the PO instead of using the Supplier's `Default Payment Template`.

### Module

buying, selling

### Version

Frappe version - 14
ERPNext version - 14

### Installation method

FrappeCloud

### Relevant log output / Stack trace / Full Error Message.

_No response_

### PR DESCRIPTION
**Pull request by Nihantra-Patel**
Version 15 and 14,

fixes: #41369

Before:

https://github.com/frappe/erpnext/assets/141945075/8bcdcce9-ca61-44ed-9508-bdf7b25bca51

After:

https://github.com/frappe/erpnext/assets/141945075/761ac532-e7fb-4960-bd31-dc18c47c882e

### CODE REVIEW

File: erpnext/selling/doctype/sales_order/sales_order.py [hunk 1]
```diff
@@ -1239,11 +1239,18 @@ def set_missing_values(source, target):
 		target.discount_amount = 0.0
 		target.inter_company_order_reference = ""
 		target.shipping_rule = ""
+		target.tc_name = ""
+		target.terms = ""
+		target.payment_terms_template = ""
```
Comment (IMS94) [inline erpnext/selling/doctype/sales_order/sales_order.py:1244]: I think the `payment_schedule` should be cleared. Only clearing the `payment_terms_template` didn't work for me. The `payment_schedule` still got copied.

**Review by IMS94 — COMMENTED** [review]
Thanks for the quick fix. Added a minor comment.

File: erpnext/selling/doctype/sales_order/sales_order.py (hunk 1 above)
Comment (Nihantra-Patel) [reply erpnext/selling/doctype/sales_order/sales_order.py:1244]: Only works properly in version 15. I think the issue will be in version 14, but I fixed it.

Comment (IMS94) [reply erpnext/selling/doctype/sales_order/sales_order.py:1244]: Thanks!

**Review by nabinhait — APPROVED** [review]

**Pull request merged**